### ISSUE TITLE
Getting `Capybara::Poltergeist::ObsoleteNode` using Phantomjs 2.1

### ISSUE THREAD

**Issue by claptimes5** [teampoltergeist/poltergeist#725]
Phantomjs 2.1 was just released, and I'm now getting the following error when trying to set an attachment on a hidden element:

```Capybara::Poltergeist::ObsoleteNode: The element you are trying to interact with is either not part of the DOM, or is not currently visible on the page (perhaps display: none is set). It's possible the element has been replaced by another element and you meant to interact with the new element. If so you need to do a new 'find' in order to get a reference to the new element.```
...
------------------
--- Caused by: ---
Capybara::Poltergeist::BrowserError:
  There was an error inside the PhantomJS portion of Poltergeist. If this is the error returned, and not the cause of a more detailed error response, this is probably a bug, so please report it.  ```

The offending line in my spec is:
`attach_file('vendor[logo]', 'spec/images/logo.png', visible: false)`

I've also specified the full path to the image and get the same failure. It appears to be failing on this command https://github.com/teampoltergeist/poltergeist/blob/master/lib/capybara/poltergeist/browser.rb#L344:

```
"{"id":"832a4709-2c86-4c8c-b143-003c2b9bf85a","name":"select_file","args":[4,4,"/my/full/path/spec/images/logo.png"]}"
```

This works in Phantomjs 1.9.8. I'm using Poltergeist 1.8.1.

If this should be reported as a regression agains Phantomjs, I'm happy to submit it there.

**Comment by claptimes5** [comment]
It looks like this is the culprit:
https://github.com/teampoltergeist/poltergeist/blob/master/lib/capybara/poltergeist/client/browser.coffee#L176

If I remove the if statement and just return `@current_command.sendResponse(true)`, my spec passes.
FileUpload was fixed in Phantomjs 2.1 so this maybe something that was used in an attempt to get things working in 2.0? 

I'm happy to make a pull request if you think this is a viable option.

**Comment by route** [comment]
I think if 2.1 is really really good, we can drop support for 1.9.x because it has so many problems with modern web.

**Comment by twalpole** [comment]
I'm taking a look at this - adding 2.1 testing, etc - Note that the actual cause of this error is that you are attempting to set a value to a non-visible field which is very driver specific to poltergeist and I don't believe works on any other driver (and shouldn't according to the webdriver spec).

**Comment by claptimes5** [comment]
Thanks... And yeah, I don't the way it's working either, but I'm using
https://github.com/blueimp/jQuery-File-Upload for ajax file uploads and am
having issues using the visible element. Would be open to suggestions on
that front, but that wasn't the intention of this thread.

**Comment by twalpole** [comment]
@claptimes5 it's a common issue file boxes since they're so often hidden to allow styling, and I wish the webdriver spec took that into account.  Basically to follow the spec (and be cross driver compatible with your tests) you need to use execute_script to adjust the CSS of the file input so that it becomes visible on the page and then call attach_file on it.  Not a great solution, but it is what it is.

**Comment by twalpole** [comment]
Ok - so I'm seeing 2 test failures with 2.1 -- attach_file and one related to javascript errors being raised to ruby errors -- once I have that fixed I'll merge it in.

**Comment by twalpole** [comment]
PR #726 should fix this (and one other small test failure) and add PhantomJS 2.1.1 to the travis test matrix.  I'll merge in as soon as the tests pass

**Comment by twalpole** [comment]
Seems like travis-ci is having issues with network connectivity currently so this may take a little longer than expected

**Comment by twalpole** [comment]
Fixed via PR #726
